src/mbgl/programs/segment.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <mbgl/gl/context.hpp>
+#include <mbgl/gl/vertex_array.hpp>
+#include <mbgl/util/optional.hpp>
+
+#include <cstddef>
+#include <vector>
+#include <map>
+
+namespace mbgl {
+
+template <class Attributes>
```

Comment:
Since there are no member functions, and the constructor is likely inlined, I don't think it will be affected. I'm hopeful this PR will reduce code size overall, since the template parameters were removed from `AttributeBinding`. I'll run bloaty and see.